processor/resourcedetectionprocessor/internal/aws/eks/detector.go

```diff
@@ -31,25 +35,42 @@ const (
 
 	// Environment variable that is set when running on Kubernetes.
 	kubernetesServiceHostEnvVar = "KUBERNETES_SERVICE_HOST"
+	authConfigmapNS             = "kube-system"
+	authConfigmapName           = "aws-auth"
 )
 
-var _ internal.Detector = (*Detector)(nil)
+type detectorUtils interface {
+	getConfigMap(ctx context.Context, namespace string, name string) (map[string]string, error)
+}
+
+type eksDetectorUtils struct {
+	clientset *kubernetes.Clientset
+}
 
 // Detector for EKS
-type Detector struct{}
+type Detector struct {
+	utils detectorUtils
+	err   error
+}
+
+var _ internal.Detector = (*Detector)(nil)
+
+var _ detectorUtils = (*eksDetectorUtils)(nil)
 
 // NewDetector returns a resource detector that will detect AWS EKS resources.
 func NewDetector(_ component.ProcessorCreateSettings, _ internal.DetectorConfig) (internal.Detector, error) {
-	return &Detector{}, nil
+	utils, err := newK8sDetectorUtils()
+	return &Detector{utils: utils, err: err}, nil
 }
 
 // Detect returns a Resource describing the Amazon EKS environment being run in.
 func (detector *Detector) Detect(ctx context.Context) (resource pdata.Resource, schemaURL string, err error) {
 	res := pdata.NewResource()
 
-	// Check if running on k8s.
-	if os.Getenv(kubernetesServiceHostEnvVar) == "" {
-		return res, "", nil
+	//Check if running on EKS.
+	isEKS, err := isEKS(ctx, detector.utils)
+	if !isEKS {
+		return res, "", err
```

Comment:
Consider adding a debug log here indicating that the detector only works on EKS.